pyproject.toml

```diff
@@ -0,0 +1,50 @@
+[tool]
+[tool.poetry]
+name = "maestrowf"
+version = "1.1.9.dev0"
+description = "A tool to easily orchestrate general computational workflows both locally and on supercomputers."
+license = "MIT License"
+classifiers = ["Development Status :: 5 - Production/Stable", "Operating System :: Unix", "Operating System :: MacOS :: MacOS X", "Intended Audience :: Developers", "Intended Audience :: Education", "Intended Audience :: Science/Research", "Topic :: Scientific/Engineering", "Topic :: System :: Distributed Computing", "Programming Language :: Python :: 3", "Programming Language :: Python :: 3.5", "Programming Language :: Python :: 3.6", "Programming Language :: Python :: 3.7"]
+homepage = "https://github.com/llnl/maestrowf"
+authors = ["Francesco Di Natale <dinatale3@llnl.gov>", "Francesco Di Natale"]
+readme = "README.md"
+packages = [
+    { include = 'maestrowf'}
+]
+
+[tool.poetry.dependencies]
+python = "*"
+chainmap = {python = "<3", version = "*"}
```

Comment:
We're dropping support for 2.7, so the `chainmap` dependency can go. I'm looking to see how we limit Python versions, but I'm not seeing that currently other than in the `classifiers`.

Reply:
That's the one piece I haven't worked out as of yet.

Reply:
I think this is the last bit, I'll approve it for merge once we figure this out.

Reply:
Oh, actually -- doing some reading, we might need to include data files through the `toml` file too..

Reply:
> That's the one piece I haven't worked out as of yet.

I found some projects that include a `pyproject.toml` [here](https://github.com/carlosperate/awesome-pyproject#project-templates) and it seems you can specify `python` similar to below.

```
[tool.poetry.dependencies]
python = "^3.7"
importlib_metadata = {version = "^1.6.0", python = "<3.8"}
typer = {extras = ["all"], version = "^0.2.1"}
rich = "^2.2.2"
pre-commit = "^2.5.1"
```

Reply:
I double checked the build and it pulls in the data file.

Reply:
> Oh, actually -- doing some reading, we might need to include data files through the `toml` file too..

I double checked the build and it pulls in the data file.

Reply:
Just to be clear we are supporting python >= 3.5 correct?

Reply:
Yep -- that's right. I reformatted some stuff last night and add `'>3.4' for the `python` line and it ran super slow in WSL. It looks like it successfully finished in the middle of the night. Here's what I tested:

```
[tool]
[tool.poetry]
name = "maestrowf"
version = "1.1.9.dev0"
description = "A tool to easily orchestrate general computational workflows both locally and on supercomputers."
license = "MIT License"
classifiers = [
    "Development Status :: 5 - Production/Stable",
    "Operating System :: Unix", "Operating System :: MacOS :: MacOS X",
    "Intended Audience :: Developers", "Intended Audience :: Education",
    "Intended Audience :: Science/Research", "Topic :: Scientific/Engineering",
    "Topic :: System :: Distributed Computing",
    "Programming Language :: Python :: 3",
    "Programming Language :: Python :: 3.5",
    "Programming Language :: Python :: 3.6",
    "Programming Language :: Python :: 3.7",
    "Programming Language :: Python :: 3.8",
]
homepage = "https://github.com/llnl/maestrowf"
documentation = "https://maestrowf.readthedocs.io/"
authors = [
    "Francesco Di Natale <dinatale3@llnl.gov>", "Francesco Di Natale"
]
maintainers = [
    "Francesco Di Natale <dinatale3@llnl.gov>", "Francesco Di Natale"
]
readme = "README.md"
packages = [
    { include = 'maestrowf'}
]

[tool.poetry.dependencies]
python = ">3.4"
coloredlogs = "*"
dill = "*"
filelock = "*"
jsonschema = ">=3.2.0"
pyyaml = ">=4.2b1"
six = "*"
tabulate = "*"

[tool.poetry.dev-dependencies]
fabric = "*"
coverage = "*"
sphinx_rtd_theme = "*"
sphinx = "*"
flake8 = "*"
pydocstyle = "*"
pylint = "*"
tox = "*"
pytest = "*"
pytest-cov = "*"
pre-commit = "*"
tox-travis = "*"
tox-pyenv = "*"


[tool.poetry.plugins."console_scripts"]
"maestro" = "maestrowf.maestro:main"
"conductor" = "maestrowf.conductor:main"


[build-system]
requires = ["poetry>=0.12"]
build-backend = "poetry.masonry.api"


[tool.poetry.urls]
"Bug Tracker" = "https://github.com/LLNL/maestrowf/issues"
"Discussions" = "https://github.com/LLNL/maestrowf/discussions"
```

Reply:
Relatedly, I listed myself as a maintainer -- feel free to add yourself @kcathey , I was planning to look up the right emails. @jwhite242 @jsemler -- want to be added?